File: polimorfismo/hr-csharp/Course.OOP.Polimorfism.HR/Entities/Employee.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Course.OOP.Polimorfism.HR.Entities
{
    public class Employee
    {
        public int EmployeeID { get; set; }

        public string FirstName { get; set; }

        public string LastName { get; set; }

        public DateTime StartDate { get; set; }

        public double BaseSalary { get; set; }


        public int CalculateWorkingDays(DateTime endDate) {
            Console.WriteLine("Método con parámetros");
            int days = Math.Abs((endDate - StartDate).Days);
            return days;
        }

        public int CalculateWorkingDays() {
            Console.WriteLine("Método sin parámetros");
            int result = CalculateWorkingDays(DateTime.Now);
            return result;
        }

        public virtual double CalculateNetSalary() {

            Console.WriteLine("Estamos en la clase Employee");

            const double socialSecurity = 0.08;

            double netSalary = BaseSalary * (1 - socialSecurity);
            return netSalary;

        }


    }
}

File: polimorfismo/hr-java/src/main/java/course/oop/polimorfism/entities/Employee.java
package course.oop.polimorfism.entities;

import java.time.Duration;
import java.time.LocalDate;

public class Employee {

    private int employeeID;
    private String firstName;
    private String lastName;
    private LocalDate startDate;
    private double baseSalary;

    public Employee(int employeeID, String firstName, String lastName, LocalDate startDate, double baseSalary) {
        this.employeeID = employeeID;
        this.firstName = firstName;
        this.lastName = lastName;
        this.startDate = startDate;
        this.baseSalary = baseSalary;
    }

    public int calculateWorkingDays(LocalDate endDate) {
        System.out.println("Método con parámetros");

        int days = (int)Duration.between(startDate.atStartOfDay(), endDate.atStartOfDay()).toDays();
        return days;
    }

    public int calculateWorkingDays() {
        System.out.println("Método sin parámetros");
        int result = calculateWorkingDays(LocalDate.now());
        return result;
    }

    public double calculateNetSalary() {

        System.out.println("Estamos en la clase Employee");

        final double socialSecurity = 0.08;

        double netSalary = baseSalary * (1 - socialSecurity);
        return netSalary;

    }
}
